### PR TITLE
#9859 fixed wrong column width calculation with colSpan

### PR DESCRIPTION
**Pull request by Holdwyn**
I've adopted the existing solution for rowHeights with rowSpan.

### CODE REVIEW

**Comment by oetiker** [comment]
are there no tests for the grid layout?

**Comment by Holdwyn** [comment]
> are there no tests for the grid layout?

of corse there are ... I've added a test case

File: framework/source/class/qx/test/ui/layout/Grid.js [hunk 1]
```diff
@@ -89,6 +89,26 @@ qx.Class.define("qx.test.ui.layout.Grid",
...
+    testColSpanWithoutFlex : function() {
+      // test with spacing
+      this._gridLayout.setSpacingX(6);
+
+      var w1 = new qx.test.ui.layout.LayoutItem(100, 100);
+      this._gridWidget.add(w1, {row: 1, column: 0});
+
+      var w2 = new qx.test.ui.layout.LayoutItem(100, 100);
+      this._gridWidget.add(w2, {row: 1, column: 1});
+
```
Comment (oetiker) [inline framework/source/class/qx/test/ui/layout/Grid.js]:
```suggestion
      this.flush();
      this.assertEquals(200, this._gridWidget.bounds.width);
```

an this too ?

File: framework/source/class/qx/ui/layout/Grid.js [hunk 2]
```diff
@@ -994,14 +994,47 @@ qx.Class.define("qx.ui.layout.Grid",
...
+            {
+              offset = colIncrements[widgetColumn+j] ? colIncrements[widgetColumn+j].offset : 0;
+              colWidths[widgetColumn+j].width += offset;
+            }
+          // col is too small and we have no flex value set
+          } else {
+            var totalSpacing = hSpacing * (widgetProps.colSpan - 1);
+            var availableWidth = hint.width - totalSpacing;
+            
+            // get the col height which every child would need to share the
```
Comment (derrell) [inline framework/source/class/qx/ui/layout/Grid.js]: get the col **width**?

**Review by derrell — CHANGES_REQUESTED** [review]
Just one typo. Looks good, I think. Thanks.

**Review by hkollmann — APPROVED** [review]

**Review by cboulanger — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Comment by level420** [comment]
5 approvals, merging, thank you @Holdwyn

**Pull request merged**